rewrite-xml/src/main/java/org/openrewrite/xml/XPathMatcher.java

```diff
@@ -164,36 +157,42 @@ public boolean matches(Cursor cursor) {
                 }
             }
 
-            return startsWithSlash || path.size() - pathIndex <= 1;
+            // we have matched the whole XPath, and it does not start with the root
+            return true;
         } else {
             Collections.reverse(path);
 
             // Deal with the two forward slashes in the expression; works, but I'm not proud of it.
-            if (expression.contains("//") && !expression.contains("://") && Arrays.stream(parts).anyMatch(StringUtils::isBlank)) {
```

Comment:
Removing `://` does not seem to break anything, but fixes XPaths with both `//` and a namespace URL condition.

I don’t think the first condition is still relevant either, however the `isBlank` check is inconsistent with the `indexOf("")` on the next line. I think it should use `isEmpty()` like I did to compute `tagMatchingParts`.

Note that the first condition is insufficient on its own because of URLs again – or actually any value that could contain double slashes in a condition.